### PR TITLE
Use empty() instead of size() to check whether we have elements in th…

### PR DESCRIPTION
**Pull request by jwillemsen**
…e map and fixed some incorrect logging

    * rmw_connext_shared_cpp/src/names_and_types_helpers.cpp:
    * rmw_connext_shared_cpp/src/service_names_and_types.cpp:
    * rmw_connext_shared_cpp/src/topic_names_and_types.cpp:
    * rmw_connext_shared_cpp/src/types/custom_data_reader_listener.cpp:

### CODE REVIEW

File: rmw_connext_shared_cpp/src/topic_names_and_types.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ get_topic_names_and_types(
   node_info->subscriber_listener->fill_topic_names_and_types(no_demangle, topics);
 
   // Copy data to results handle
-  if (topics.size() > 0) {
+  if (!topics.empty() > 0) {
```
Comment (dirk-thomas) [inline rmw_connext_shared_cpp/src/topic_names_and_types.cpp]: Incomplete change - mixing book and integer comparison.

Comment (jwillemsen) [reply rmw_connext_shared_cpp/src/topic_names_and_types.cpp]: Fixed

**Comment by jwillemsen** [comment]
Found these when working on rmw_opendds (see https://github.com/jwillemsen/rmw_opendds/tree/jwi-openddsros2)

**Comment by tfoote** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5887/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2435/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4880/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5788/)

**Comment by jwillemsen** [comment]
Jenkins errors don't look to be related to my changes

Linux:
<pre>
04:49:06 ==> vcs import "src" --force --retry 5 --input 01-rmwc.repos
04:49:06 === src\ros2/rmw_connext (git) ===
04:49:06 Invocation of command 'import' on client 'git' failed: PermissionError: [WinError 5] Access is denied: 'src\\ros2/rmw_connext' (c:\python37\lib\site-packages\vcstool\clients\git.py:209)
04:49:06 <== 'vcs import "src" --force --retry 5 --input 01-rmwc.repos' exited with return code '1'
</pre>

Windows:
<pre>
02:33:56 -- Found ament_lint_auto: 0.6.2 (/home/rosbuild/ci_scripts/ws/install/ament_lint_auto/share/ament_lint_auto/cmake)
02:33:56 CMake Error at /home/rosbuild/ci_scripts/ws/install/rmw_implementation_cmake/share/rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake:60 (message):
02:33:56   Could not find ROS middleware implementation 'NOTFOUND'.  Choose one of the
02:33:56   following:
02:33:56 Call Stack (most recent call first):
02:33:56   CMakeLists.txt:28 (get_default_rmw_implementation)
</pre>

**Comment by sloretz** [comment]
Checking CI again since it's been a while
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5962/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2484/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4928/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5851/)

**Comment by sloretz** [comment]
Linux CI failure looks unrelated, will wait before rerunning

```
/home/rosbuild/ci_scripts/venv/bin/python: can't open file '/home/rosbuild/ci_scripts/venv/bin/vcs': [Errno 2] No such file or directory
```

**Comment by sloretz** [comment]
CI using ros2/ci#234

* Linux [Build Status](https://ci.ros2.org/job/ci_linux/5964/)
* Linux aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/)

**Comment by sloretz** [comment]
CI looks good, thanks for the PR!

**Pull request merged**